pvlib/tests/iotools/test_pvgis.py

```diff
@@ -4,12 +4,280 @@
+    (testfile_radiation_csv, 'expected_radiation_csv',
+     args_radiation_csv, False, url_hourly_radiation_csv),
+    (testfile_radiation_csv, 'expected_radiation_csv_mapped',
+     args_radiation_csv, True, url_hourly_radiation_csv),
+    (testfile_pv_json, 'expected_pv_json', args_pv_json, False, url_pv_json),
+    (testfile_pv_json, 'expected_pv_json_mapped', args_pv_json, True,
+     url_pv_json)])
+def test_get_pvgis_hourly(requests_mock, testfile, expected_name, args,
+                          map_variables, url_test, request):
+    """Test that get_pvgis_hourly generates the correct URI request and that
+    _parse_pvgis_hourly_json and _parse_pvgis_hourly_csv is called correctly"""
+    # Open local test file containing McClear mothly data
+    with open(testfile, 'r') as test_file:
+        mock_response = test_file.read()
+    # Specify the full URI of a specific example, this ensures that all of the
+    # inputs are passing on correctly
+    requests_mock.get(url_test, text=mock_response)
+    # Make API call - an error is raised if requested URI does not match
+    out, inputs, metadata = get_pvgis_hourly(
+        latitude=45, longitude=8, map_variables=map_variables, **args)
+    # Get expected dataframe from fixture
+    expected = request.getfixturevalue(expected_name)
+    # Compare out and expected dataframes
+    assert_frame_equal(out, expected)
+
+
+def test_get_pvgis_hourly_bad_status_code(requests_mock):
+    # Test if a HTTPError is raised if a bad request is returned
+    requests_mock.get(url_pv_json, status_code=400)
+    with pytest.raises(requests.HTTPError):
+        get_pvgis_hourly(latitude=45, longitude=8, **args_pv_json)
+    # Test if HTTPError is raised and error message is returned if avaiable
+    requests_mock.get(url_pv_json, status_code=400,
+                      json={'message': 'peakpower Mandatory'})
+    with pytest.raises(requests.HTTPError):
+        get_pvgis_hourly(latitude=45, longitude=8, **args_pv_json)
+
+
+url_additional_inputs = 'https://re.jrc.ec.europa.eu/api/seriescalc?lat=55.6814&lon=12.5758&outputformat=csv&angle=0&aspect=0&pvcalculation=1&pvtechchoice=crystSi&mountingplace=free&trackingtype=0&components=1&usehorizon=1&optimalangles=1&optimalinclination=1&loss=2&userhorizon=10%2C15%2C20%2C10&peakpower=5'  # noqa: E501
+
+
+def test_get_pvgis_hourly_additional_inputs(requests_mock):
+    # Test additional inputs, including userhorizons
+    # Necessary to pass a test file in order for the parser not to fail
+    with open(testfile_radiation_csv, 'r') as test_file:
+        mock_response = test_file.read()
+    requests_mock.get(url_additional_inputs, text=mock_response)
+    # Make request with userhorizon specified
+    get_pvgis_hourly(
+        latitude=55.6814, longitude=12.5758, outputformat='csv',
+        usehorizon=True, userhorizon=[10, 15, 20, 10], pvcalculation=True,
+        peakpower=5, loss=2, trackingtype=0, components=True,
+        optimalangles=True)
```

Comment:
Does the output here need to be checked?  Or is this just to verify that no error is raised?

Reply:
I'm not sure I understand the question?

Reply:
The purpose of this test function is solely to check that the input parameters not tested in the previous test are formatted correctly in the request.

The output would not change significantly, hence I do not seen any additional benefit in adding a third test file.

Reply:
Thanks, I forgot that requests-mock will raise an error if you try to fetch an unexpected URL.  Makes sense.